gtsfm/runner/gtsfm_runner_base.py

```diff
@@ -202,6 +211,24 @@ def run(self) -> None:
         process_graph_generator.save_graph()
 
         pairs_graph = self.retriever.create_computation_graph(self.loader)
+
+        if self.parsed_args.run_visualizer:
+            image_vis_data = {}
+            for i in range(len(self.loader)):
+                io_utils.save_image(
+                    self.loader.get_image(i),
+                    os.path.join(REACT_METRICS_PATH, f"{i}.png")
+                )
+                image_vis_data[i] = {
+                    "shape": self.loader.get_image_shape(i),
+                    "focal_length": self.loader.get_camera_intrinsics(i).fx()
+                }
+
+            io_utils.save_json_file(
+                os.path.join(REACT_METRICS_PATH, "image_shapes.json"),
+                image_vis_data
+            )
```

Comment:
why do we need to save them again?

Reply:
so we can retrieve the images and shapes easily in the JS visualization

Reply:
I get why you are creating the json file, but why duplicate the images? you could add the original image filename in the json too, and just open that image? am I missing something about reading images from the dataset?